code/src/pgsql_io/pgsql_io.ml

```diff
@@ -961,6 +1116,36 @@ and create_sm_login ?passwd ~user t =
   let open Abbs_future_combinators.Infix_result_monad in
   Io.wait_for_frames t >>= fun frames -> create_sm_process_login_frames ?passwd ~user t frames
 
+(* TODO: Maybe it's worth aglutinating step_02 and step_03
+   under the same recursive function, left it for a future
+   refactor *)
+and create_sm_scram_sha256_step_03 ?passwd ~user client_final t =
+  let open Pgsql_codec.Frame.Backend in
+  let open Abbs_future_combinators.Infix_result_monad in
+  Io.wait_for_frames t
+  >>= function
+  | AuthenticationSASLFinal { data } :: fs -> (
+      match Auth_scram.verify_server_final client_final data with
+      | Ok _ ->
+          (* Should be waiting for the next frame as an AuthenticationOk *)
+          create_sm_process_login_frames ?passwd ~user t fs
+      | _ -> Abb.Future.return (Error `Unsupported_auth_sasl_err))
+  | fs -> Abb.Future.return (Error (`Unmatching_frame fs))
+
+and create_sm_scram_sha256_step_02 ?passwd ~user client_request t =
+  let open Pgsql_codec.Frame.Backend in
```

Comment:
This change is not a requirement, I'm just expressive how I would do it.  If you want a shorter name, prefer to re-bind the module with `let module B = Pgsql_codec.Frame.backend`.  I know that this seems arbitrary and contradictory given a few lines down where the module is opened.  It is arbitrary and contradictory!  The difference, in my head, is that here we're importing a WHOLE module for 1 symbol.  So being explicit wins.  A few lines down when we open the module, it's because we are accessing a bunch of symbols from it, so being explicit actually gets in the way of reading the code.  This is obviously in the eye of the beholder.